graph/graph.go

```diff
@@ -413,6 +468,17 @@ func (graph *Graph) ByParent() (map[string][]*image.Image, error) {
 	return byParent, err
 }
 
+// If the images and layers are in pulling chain, retain them.
+// If not, they may be deleted by rmi with dangling condition.
+func (graph *Graph) Retain(layerIDs []string, imgID string) {
+	graph.retained.Add(layerIDs, imgID)
+}
+
+// Unretain removes the referenced image id from the provided set of layers.
+func (graph *Graph) Unretain(layerIDs []string, imgID string) {
```

Comment:
Any reason why the word `Unretain` was chosen? `Retain` and `Release` seems more natural and is already widely used by other reference counting systems.